src/test/java/ru/yandex/clickhouse/integration/RowBinaryStreamTest.java

```diff
@@ -150,6 +206,77 @@ public void writeTo(ClickHouseRowBinaryStream stream) throws IOException {
         Assert.assertEquals(rs.getDouble("float32"), 123.456);
         Assert.assertEquals(rs.getDouble("float64"), 42.21);
 
+        final Date[] dateArray = (Date[]) rs.getArray("dateArray").getArray();
+        Assert.assertEquals(dateArray.length, dates1.length);
+        for (int i = 0; i < dateArray.length; i++) {
+            // expected is Date at start of the day in server timezone
+            DateTime dt = new DateTime(dates1[i].getTime())
```

Comment:
Yes, in fact having discussed this with clickhouse developers and checked the documentation and implementations of other drivers, I consider that the current behavior with dates that I described in my previous comment incorrect.
I have a PR to fix that https://github.com/yandex/clickhouse-jdbc/pull/120
You can make a quick review of it if you'd like. I plan to merge it in a couple of days.
Then the tests in this PR need to be fixed.